Demo/src/main/java/com/braintreepayments/demo/VenmoFragment.java

```diff
@@ -35,14 +37,20 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
         venmoButton = view.findViewById(R.id.venmo_button);
         venmoButton.setOnClickListener(this::launchVenmo);
 
-        braintreeClient = getBraintreeClient();
-        venmoClient = new VenmoClient(this, braintreeClient);
-        venmoClient.setListener(this);
+        venmoLauncher = new VenmoLauncher(this, venmoAuthChallengeResult ->
+                venmoClient.tokenizeVenmoAccount(venmoAuthChallengeResult, this::handleVenmoResult));
```

Comment:
More 🧠 🌦️ : I wonder if composing a reference to `BraintreeClient` within each Launcher would be a good design pattern once `ClientTokenProvider` is no longer required. It would make a method like `venmoLauncher.authenticateVenmoUser()` possible. It might be cool, but it might be worth holding off to keep it simple.

Reply:
Agreed I have a separate ticket to do this across all the modules - I think it would make most merchant integration stories simpler and we can add a params object for customization if needed